### DISCUSSION TITLE
Don't hang on to failed-load activities buffers

### DISCUSSION THREAD

**Discussion by jdtsmith** [alphapapa/activities.el#112] [Ideas]
The magit bookmark discussion (#111) reminded me: I find it odd that activities stores its own "warning, we couldn't load this buffer" buffer in its recent buffer list.  So if you load an activity, see a failed buffer restore, and then close that frame/tab, on the next load that you get a "meta" "we can't load the buffer that mentions how you couldn't load some other buffer", and so on. 

I'd expect such activities-generated buffers to be omitted on save.

**Comment by alphapapa** [comment]
If we omitted those buffers, what would we display in their stead?

**Reply by jdtsmith** [reply]
Nothing.  One reminder that a mode needs bookmark support is enough I think.

**Reply by alphapapa** [reply]
It's not possible to display nothing in a window.  Every window must display a buffer.

**Reply by jdtsmith** [reply]
I mean don't restore such a window at all.

**Reply by alphapapa** [reply]
That would mean corrupting the window configuration that was saved.  I don't see how that would be an improvement.

**Reply by jdtsmith** [reply]
And if it's the only window...
  
How about giving the "cannot restore" buffer that activities generates bookmark support, so at least you only get one version?

**Reply by alphapapa** [reply]
I don't feel like that would be worth the trouble.  These buffers are meant to be dismissed anyway.  At least, such an idea would be very near the bottom of my Emacs-related to-do list.

**Reply by alphapapa** [reply]
@jdtsmith Having said that, I'm not opposed to, e.g. making an option for a function to return the buffer to be shown in such cases, so users could configure it to be whatever they want (e.g. blank).  But I don't have time to work on that myself right now.  Patches welcome, though.

**Reply by jdtsmith** [reply]
No worries, can put it on the back burner and see if it bothers anyone else.
